src/Auth/HttpHeaderAuthTrait.php

```diff
@@ -0,0 +1,33 @@
+<?php
+namespace Yiisoft\Yii\Web\Auth;
+
+use Psr\Http\Message\ServerRequestInterface;
+
+trait HttpHeaderAuthTrait
```

Comment:
I'd avoid using trait in this case. It hides complexity but doesn't help solving it.

Reply:
Inheritance suits it better in this case...